FrameworkSystem/private/standardLogging/gLogging.py

```diff
@@ -0,0 +1,370 @@
+      self._level = LogLevels.getLevelValue(father.getLevel())
+    else:
+      self._options = {'showHeaders': True, 'showThreads': False, 'Color': False, 'Path': False}
+      # the native level is not used because it has to be to debug to send all messages to the log central
+      self._level = None
+
+    # name of the gLogging
+    self.name = name
+
+    # dictionary of the option state, modified by the user or not
+    # this is to give to the options the same behaviour that the logging level:
+    # - propagation from the parent to the children when their levels are not set by the developer himself
+    # - stop the propagation when a developer set a level to a child
+    self._optionsModified = {'showHeaders': False, 'showThreads': False}
+    self._levelModified = False
+
+    self._backendsList = []
+
+    self._logger = logging.getLogger(fathername).getChild(name)
+
+  def showHeaders(self, yesno=True):
+    """
+    Depending on the value, display or not the prefix of the message.
+    :params yesno: boolean determining the behaviour of the display
+    """
+    self._setOption('showHeaders', yesno)
+
+  def showThreadIDs(self, yesno=True):
+    """
+    Depending on the value, display or not the thread ID.
+    :params yesno: boolean determining the behaviour of the display
+    """
+    self._setOption('showThreads', yesno)
+
+  def _setOption(self, optionName, value):
+    """
+    Depending on the value, modify the value of the option.
+    This option will not be modified anymore. 
+    The options of the children will be updated if they were not modified before by a developer
+    :params optionName: string representing the name of the option to modify
+    :params value: boolean to give to the option  
+    """
+    self._options[optionName] = value
+    self._optionsModified[optionName] = True
+    self._setChildrenDisplayOptions(optionName, self._options)
+    # update the format to apply the option change
+    self._updateFormat()
+
+  def registerBackends(self, desiredBackends, backendOptions=None):
+    """
+    Attach a list of backends to the gLogging object.
+    :params desiredBackends: a list of different names attaching to differents backends.
+                             example : ['stdout', 'file', 'server']
+    :params backendOptions: a dictionary of different backend options. 
+                            example: {'FileName': '/tmp/log.txt'}
+    """
+    for stringHandler in desiredBackends:
+      stringHandler = stringHandler.strip().lower()
+
+      # check if the name is correct
+      if stringHandler in self._backendsDict:
+        backend = self._backendsDict[stringHandler]
+
+        # check if the backend is already in the list
+        if backend not in self._backendsList:
+          if backendOptions is not None:
+            # give a copy to avoid that the backends modify the dictionary
+            backend.setParameters(backendOptions.copy())
```

Comment:
It is a bit counter intuitive to have to set the parameters, and then call configure so that the parameter is applied. Was it like this in the past already ?